### DISCUSSION TITLE
v3.4 - New climate page and some bug fixes

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#877] [Announcements]
### As always, we would like to thank everyone who donated to this project and remind you that every donation helps to support this project - so if you haven't made a donation yet, now would be a good time ;)

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;
## General
It's been a while since the last update - but we also need time for our families.
In this version we have reprogrammed the climate page and the code from scratch and believe me, climate is really no fun :)
The Home page also got some attention and is loading much faster.
In addition, we have fixed some bugs and adjusted small things in the UI design.
### 

&nbsp;
## Updating
Since in this update lots of input to the blueprint changed, we highly recommend you review your settings and make sure all fields have the correct selection. We did our best to support your legacy settings and avoid breaking your system, but please double check your settings if you see something not working as expected.

### Files that need to be reloaded:
1. nspanel_blueprint.yaml - v3.4
2. nspanel_eu.tft or nspanel_us.tft - v3.4
3. nspanel_esphome.yaml - v3.4

### Instructions
 _You can find the update procedures following here:_
- [(EN) HowTo - All important thing you should know - Update](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#2-update-blueprint)
- [(DE) HowTo - Alle wichtigen Dinge die man wissen sollte - Update](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(DE)-HowTo---Alle-wichtigen-Dinge-die-man-wissen-sollte#2-update-blueprint)

&nbsp;
## Breaking changes
- Customization of climate icon on home page (chip 03) - Now it follows the standard icon & color selection used on page climate, but with a thermometer-lines for heating and a thermometer for idle.

&nbsp;
## Overview of all changes
1. New Climate page for EU and US
2. More flexible date and time engine (#744)
3. Users now can select alignment on entities pages (#753)
4. Auto-update of entities pages (#819)
5. Much faster load of home page

&nbsp;
### 1. New Climate Page
The climate page was totally rebuild and now is more responsive, with more information available and much easier to use.
The Climate Page pulls all information from the selected Climate Device in the Blueprint. 
This allows us to also support climates like sauna or with Fahrenheit.

![2023-06-18 21 37 13](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/41958506/d558b552-e31d-4272-bce9-e65eacde6c36)

- Easy access buttons for controlling the climate entity.
- 4 user selectable values shown
- 2 user defined buttons for more flexibility

### 2. More flexible date & time engine
Apart of the options for date and time you are already familiar, users can now define their preferred format using any of the options available for [Strftime in ESPHome](https://esphome.io/components/time/index.html#strftime).
To use your own format, just type it in the drop down:
![IMG_6276](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/0a9b7f2d-d90b-4860-bd7c-fc49febb7b88)

Please be careful using this, as you can easily exceed the limited space for time and date strings in your panel.

### 3. Select alignment on entities pages
You can now select how the column showing the values will be aligned in your entities pages:
![IMG_4109](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/e4e8f605-a64b-4ad4-802c-1e7e055a8f0d)

### 4. Auto-update of entities pages
Now while an Entity page is visible, all the entities will be updated when it's statuses changes, improving the user's experience.

### 5. Much faster load of home page
The Home page is now updated in the background, even when the panel is sleeping, and will be shown almost instantaneously with a touch in the screen or when navigating from other pages.

&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

## What's Changed
* fix: corrects date_format by @albinmedoc in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/568
* Update the correct Nextion variable for sleep modus by @doron1 in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/667

## New Contributors
* @albinmedoc made their first contribution in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/568
* @doron1 made their first contribution in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/667

**Full Changelog**: https://github.com/Blackymas/NSPanel_HA_Blueprint/compare/v3.2.2...v3.3

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v3.4'>v3.4 - New climate page and some bug fixes</a>.</em>

**Comment by Doughboy68** [comment]
New climate works great and is useable now!!! Previous versions had disabled.
Is it possible to add more than one climate entity and toggle/cycle between them somehow? I notice that if I update the blueprint it is pretty instant updating to new climate entity (eg. between my ecobee and zwave fireplace thermostat) so maybe this is possible?

The alignment for entities seems backwards to me... left and right should be swapped.

**Reply by edwardtfn** [reply]
You can assign multiple climate entities to the button pages and then open each one with a long press in its button. Is this what you are looking for?

**Reply by Doughboy68** [reply]
I did not realize could add a climate. entity to any button. I tried this and works... don't have to use a long press btw... a short one works :)

**Comment by gavilaso** [comment]
Hi Guys! i just updated to 3.4 (US) but now all i see on the screen is "Loading" i reboot it a couple of time and after the splash screen i just go back to the loading... 

any suggestions?

**Reply by gavilaso** [reply]
Nevermind, i just find out that when i installed i set it to get the tft file from home assistant.. so i forgot to upload the new one.. i'm good now thanks!!

**Comment by rasmus-wi** [comment]
As soon as I upgraded to 3.4, home assistant started being VERY slow, and occasionally restarted. Home assistant logs were full of an Nspanel related error.
`2023-06-23 17:56:57.871 WARNING (MainThread) [homeassistant.components.automation.nspanel1_config] NSPanel1 Config: Maximum number of runs exceeded
2023-06-23 17:56:58.783 WARNING (MainThread) [homeassistant.components.automation.nspanel1_config] NSPanel1 Config: Maximum number of runs exceeded
2023-06-23 17:56:59.919 WARNING (MainThread) [homeassistant.components.automation.nspanel1_config] NSPanel1 Config: Maximum number of runs exceeded
2023-06-23 17:57:00.370 WARNING (MainThread) [homeassistant.components.automation.nspanel1_config] NSPanel1 Config: Maximum number of runs exceeded`
As well as various time outs
`023-06-23 18:57:42.151 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /os/info request
2023-06-23 18:57:42.352 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /core/stats request
2023-06-23 18:57:42.354 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /supervisor/stats request`
After a lot of troubleshooting I figured out that the problem was that I had the power reading from my electricity meter on my home page, and more values from the meter on an entity page. The electricity meter pushes values to HA at a pretty high rate (1-2 readings per second), and every value that arrives seems to make the NSPanel request its config. Add in the fact that I have three of them and Home Assistant gets very flooded with requests.

The solution seemed to be not having the power readings on the home page, but only on the entity page. This still floods HA, but only while the entity page is displayed.

**Reply by edwardtfn** [reply]
Which version have you used previously?
The update of entities pages was introduced on v3.4, but the update of home page was there before, the difference ith 3.4 I that this update happens for some componentes (date, time, etc) even when the page is not visible. I have to double check about these values field. I'm pretty sure hose values are also updating on every change, but it is like this since a while ago (it haven't with v3.4).
The limit of concurrent instances was reduced from 5000 to 50 on v3.3 (#652). This could be related to the fact that you are seen this error now and also to some of the slowness, as it is now flooding your logs.

If you want to troubke hoot this, you could change thi back to 5000 by editing the [line 3496](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml#L3496) of your nspanel_blueprint.yaml (under blueprints\automations\Blakymas) to be `max: 5000` and then restart your Home Assistant (or just reload automations).
This will just hide the issue again, but if your panel then works as before, we will have the info we need to think in a way to reduce this load. I would appreciate if you can give a feedback about this proposed change.

**Comment by lloydbayley** [comment]
Love the update. Heartfelt thanks to everyone involved. SOOOOO much speedier and indications for climate etc. Thanks again! :)

**Reply by edwardtfn** [reply]
Thanks for the kind words. It means a lot. 🙂

**Reply by lloydbayley** [reply]
My pleasure. Your recent contributions have not gone unnoticed also.

**Comment by hkr086** [comment]
Do you know if it works with gree integration?

**Reply by edwardtfn** [reply]
It should work.
